components/zcash_protocol/src/consensus.rs

```diff
@@ -335,6 +335,16 @@ pub trait Parameters: Clone {
     }
 }
 
+impl<P: Parameters> Parameters for &P {
+    fn network_type(&self) -> NetworkType {
+        (*self).network_type()
+    }
+
+    fn activation_height(&self, nu: NetworkUpgrade) -> Option<BlockHeight> {
+        (*self).activation_height(nu)
+    }
+}
+
```

Comment:
I'm confused as to why this is necessary. All crates in librustzcash compile without it, and I would have thought that receiver coercion suffices in almost all cases.

Reply:
It's not strictly necessary; I was just annoyed that I had to clone a `Parameters` instance in a case where it was really unnecessary.